core/network/corda-interop-app/interop-workflows/src/main/kotlin/com/weaver/corda/app/interop/flows/AssetExchangeHTLCWrapperFlows.kt

```diff
@@ -0,0 +1,313 @@
+                
+                subFlow(LockAssetHTLC.Initiator(
+                    lockInfoData, 
+                    assetRef!!,
+                    assetStateDeleteCommand,
+                    recipient!!
+                ))
+            })
+            
+        } else {
+            println("Lock Mechanism not supported.")
+            Left(Error("Lock Mechanism not supported."))
+        }
+    } catch (e: Exception) {
+        println("Error locking asset: ${e.message}\n")
+        Left(Error("Failed to lock asset: ${e.message}"))
+    }
+}
+
+/**
+ * The ClaimAssetHTLC flow is used to claim a locked asset using HTLC.
+ *
+ * @property contractId The unique identifier for an AssetExchangeHTLCState.
+ * @property assetClaim The [AssetLocks.AssetClaimHTLC] containing hash preImage.
+ */
+
+@InitiatingFlow
+@StartableByRPC
+class ClaimAsset(
+        val contractId: String,
+        val claimInfo: AssetLocks.AssetClaim,
+        val assetStateCreateCommand: CommandData,
+        val assetStateContractId: String,
+        val updateOwnerFlow: String
+) : FlowLogic<Either<Error, SignedTransaction>>() {
+    /**
+     * The call() method captures the logic to claim the asset by revealing preimage.
+     *
+     * @return Returns SignedTransaction.
+     */
+    @Suspendable
+    override fun call(): Either<Error, SignedTransaction> = try {
+        if (claimInfo.lockMechanism == AssetLocks.LockMechanism.HTLC) {
+            val claimInfoHTLC = AssetLocks.AssetClaimHTLC.parseFrom(
+                Base64.getDecoder().decode(claimInfo.claimInfo.toByteArray())
+            )
+            val claimInfoData = AssetClaimHTLCData(
+                hashPreimage = OpaqueBytes(Base64.getDecoder().decode(claimInfoHTLC.hashPreimageBase64.toByteArray()))
+            )
+            subFlow(ClaimAssetHTLC.Initiator(
+                contractId,
+                claimInfoData,
+                assetStateCreateCommand,
+                assetStateContractId,
+                updateOwnerFlow
+            ))
+        } else {
+            println("Lock Mechanism not supported.")
+            Left(Error("Lock Mechanism not supported."))
+        }
+    } catch (e: Exception) {
+        println("Error claiming: ${e.message}\n")
+        Left(Error("Failed to claim: ${e.message}"))
+    }   
+}
+
+/**
+ * The UnlockAssetHTLC flow is used to unlock a locked asset using HTLC.
+ *
+ * @property contractId The unique identifier for an AssetExchangeHTLCState.
+ */
+
+@InitiatingFlow
+@StartableByRPC
+class UnlockAsset(
+        val contractId: String,
+        val assetStateCreateCommand: CommandData,
+        val assetStateContractId: String
+) : FlowLogic<Either<Error, SignedTransaction>>() {
+    /**
+     * The call() method captures the logic to unlock an asset.
+     *
+     * @return Returns SignedTransaction.
+     */
+    @Suspendable
+    override fun call(): Either<Error, SignedTransaction> = try {
+        subFlow(UnlockAssetHTLC.Initiator(
```

Comment:
How does UnlockAsset know that it need to call UnlockAssetHTLC.Initiator but not say UnlockAssetECDLP.Initiator ?

Reply:
Currently it doesn't know. This is something I was planning to resolve in future, as currently there is no input for lock mechanism in Unlock Transaction there's no way to find out.
This most probably would require better way to identify lock Mechanism.